docs/src/main/paradox/scala/http/routing-dsl/exception-handling.md

```diff
@@ -62,10 +62,12 @@ Note that `IllegalRequestException`s' stack traces are not logged, since instanc
 normally contain enough information to provide a useful error message.
 
 @@@ note
-Users are strongly encouraged not to rely on the `ExceptionHandler` as a means
-of handling 'normally exceptional' situations.
 
-Exceptions are known to have a negative performance impact for cases
+Users are strongly encouraged not to rely on using the `ExceptionHandler` as a means of handling errors. By errors, we mean things that are an expected part of normal operations: for example, issues discovered during input validation. The `ExceptionHandler` is meant to be a means of handling failures. See [Failure vs Error](https://www.reactivemanifesto.org/glossary#Failure) in the glossary of the [Reactive Manifesto](https://www.reactivemanifesto.org).
```

Comment:
Yeah exactly, well explained :)

Reply:
Should we point to `onComplete` and the `RejectionHandler` here as alternatives?

Reply:
@raboof yes but maybe next iteration (don't have time this week) :-)